homeassistant/helpers/intent.py

```diff
@@ -726,6 +733,15 @@ def async_can_handle(self, intent_obj: Intent) -> bool:
         """Test if an intent can be handled."""
         return self.platforms is None or intent_obj.platform in self.platforms
 
+    @cached_property
+    def effective_slot_schema(self) -> vol.Schema:
+        """Return the schema including any extra slots."""
+        if self.slot_schema is None:
+            return vol.Schema({})
+        if isinstance(self.slot_schema, vol.Schema):
+            return self.slot_schema
+        return vol.Schema(self.slot_schema)
```

Comment:
Why is this a thing? the `slot_schema` type is `vol.Schema | None`

Reply:
Although it is indeed annotated as `vol.Schema | None`, many of the integrations actually use a `dict` (see `DynamicServiceIntentHandler` as an example). Unless we want to require all intents to switch to `vol.Schema`, I chose to explicitly cast it to `vol.Schema` here.